client/blocks/login/index.jsx

```diff
@@ -1054,9 +1055,9 @@ export default connect(
 		isSecurityKeySupported: isTwoFactorAuthTypeSupported( state, 'webauthn' ),
 		linkingSocialService: getSocialAccountLinkService( state ),
 		partnerSlug: getPartnerSlugFromQuery( state ),
-		isFromAkismet: !! new URLSearchParams( getRedirectToOriginal( state )?.split( '?' )[ 1 ] )
-			.get( 'back' )
-			?.startsWith( 'https://akismet.com' ),
+		isFromAkismet: isAkismetRedirect(
```

Comment:
The GitHub bot [flagged this as insecure in former PR](https://github.com/Automattic/wp-calypso/pull/99693#pullrequestreview-2652996242) so I've implemented the suggested fix in this helper function